python/triton/language/core.py

```diff
@@ -9,9 +9,12 @@ def _to_ir(x, builder):
     if isinstance(x, bool):
         return builder.get_int1(x)
     elif isinstance(x, int):
```

Comment:
I'm thinking it would be great to specialized for both uint32 and uint64 here. The perf implications of int64 are bad enough to warrant a specialized kernel for 2**31 <= x < 2**32. And also avoiding error for very large positive int64 would be pretty valuable

Reply:
Done. I also added `test_value_specialization` and `test_value_specialization_overflow` to make sure it worked right.